filters/token-amount/filter.py

```diff
@@ -0,0 +1,123 @@
+import operator
+import spacy
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+from collections import defaultdict
+from typing import Union
+
+"""
+A filter on if the tokens contain specific keywords a certain number of times.
+"""
+
+
+class TokenAmountFilter(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
+
+    def __init__(
+        self,
+        keywords: Union[list, str] = None,
+        thresholds: Union[list, int] = None,
+        operations: Union[list, str] = None,
+    ):
+        super().__init__()
+        self.max_input_length = self.get_input_length(keywords, thresholds, operations)
+        self.final_operators = self.parse_operator(operations)
+        self.final_keywords = self.convert_scalar_to_list(keywords)
+        self.final_thresholds = self.convert_scalar_to_list(thresholds)
+        self.nlp = spacy.load("en_core_web_sm")
+        self.sanity_check()
+
+    def get_input_length(self, keywords, thresholds, operations):
+        all_inputs = [keywords, thresholds, operations]
+        lengths = defaultdict(int)
+        for key, user_input in zip(
+            ["keywords", "thresholds", "operations"], all_inputs
+        ):
+            if isinstance(user_input, list):
+                lengths[key] = len(user_input)
+            else:
+                lengths[key] = 1
+
+        unique_lengths = set(lengths.values())
+        if len(unique_lengths) == 1:
```

Comment:
I guess this is the case for inputs with all non-list, right?

The if clause should be `unique_lengths == {1}`, because if the three inputs are **all** list with the same length, len(unique_lengths) will be equal to 1 as well.

Reply:
It seems this issue does not affect the filter function, and all pytest are passed.

It would be good to fix it though :smile:.

Reply:
ha! yes indeed, good catch again.